### DISCUSSION TITLE
Are Gamepads Supported?

### DISCUSSION THREAD

**Discussion by Adriwin06** [dakanji/RefindPlus#189] [General]
Hi! Sorry if this is a dumb question, but is it technically possible to add Gamepad suport? Like if I only have my PS5 or Xbox controller plugged in, and not any keyboard/mouse, I could still select which OS I want to boot.
If it is possible, can you explain quickly how to do it, and I will learn the requiered skills to do it.
It's for a PC with SteamOS, Batocera.linux, Windows 11, and Linux. So if I want to play, I don't need to reach out to my keyboard.
Thanks in advance for the reply!

**Comment by dakanji** [comment]
Hi,

Interesting question but I am afraid I have never tried with a gamepad. 

I suppose you have tried simply connecting this and it doesn't work. If not, please do and let's know how it goes.

Chances are that it works like a mouse and that you need to activate the `enable_mouse`  config setting. As said, please get back with how you fare.

**EDIT**

Thinking about it a bit, I think it is not likely a gamepad will work in the uEFI space and that it most likely requires drivers loaded by the OS after this is booted.

Might be that the device you want to use it on has these in the firmware and they are already loaded for use but this is a bit of a long shot and not typical. 

If it relies on OS Loaded drivers, the only way to make it work before booting an OS would be to either amend the firmware to load such or get hold of a driver that provides support which RefindPlus can load.

Not 100% sure but would expect this to be the case. Either way, no harm in trying as discussed earlier

**Comment by Adriwin06** [comment]
Thanks you really much for the detailed reply!
Unfortunately I can't try I'm on vacation and there must have been a misunderstanding since I haven't set up a bootloader right now, I'm searching one to replace my disk swap boot, and I think that this one is the best one.

And to reply to your edit, if Gamepad isn't possible, do you think it is possible to make  a remote control to select the OS trough rEFInd, and make it act like a keyboard for it?

**Comment by dakanji** [comment]
Well, not saying gamepad is not possible but that it is unlikely to work just like that. Needs to be tested to confirm though. Same basically applies to any others. I suppose since touch works, other non standard devices might also do. All depends on drivers.
